packages/api/tests/lib/MessageTemplateStore.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const test = require('ava');
+const { recursivelyDeleteS3Bucket, s3 } = require('@cumulus/common/aws');
+const { randomString } = require('@cumulus/common/test-utils');
+
+const MessageTemplateStore = require('../../lib/MessageTemplateStore');
+
+let bucket;
+test.before(async () => {
+  bucket = randomString();
+  await s3().createBucket({ Bucket: bucket }).promise();
+});
+
+test.beforeEach(async (t) => {
+  t.context.workflowName = randomString();
+  t.context.stackName = randomString();
+
+  t.context.messageTemplateStore = new MessageTemplateStore({
+    bucket,
+    name: t.context.name,
+    stackName: t.context.stackName,
+    s3: s3()
+  });
+});
+
+test.after.always(async () => {
+  await recursivelyDeleteS3Bucket(bucket);
+});
+
+const requiredParams = ['bucket', 's3', 'stackName'];
+
+requiredParams.forEach((requiredParam) => {
+  test(`The MessageTemplateStore constructor requires that ${requiredParam} be set`, async (t) => {
+    const { stackName } = t.context;
+
+    const params = {
+      bucket,
+      s3: s3(),
+      stackName
+    };
+    delete params[requiredParam];
+
+    try {
+      new MessageTemplateStore(params);
+      t.fail('Expected an error to be thrown');
+    }
+    catch (err) {
```

Comment:
a more succinct way to write this try-catch block might be to use `.throws`. [docs](https://github.com/avajs/ava/blob/master/docs/03-assertions.md#throwsfn-expected-message)

Reply:
In theory that's true, but there are issues when using `.throws()` with an async function.  The 1.x branch of ava has added a `.throwsAsync()` function to better handle a case like this, but 1.x is still pre-release and we're using 0.25.

Reply:
Ah, that's good to know. Thank you for the explanation.